File: app/src/main/java/kr/co/jayhy/weather/repo/WeatherCurrentRepo.java
package kr.co.jayhy.weather.repo;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by jhkim on 2017-08-06.
 */

public class WeatherCurrentRepo {

    @SerializedName("cod")
    public String cod;

    @SerializedName("coord")
    public Coord coord;

    @SerializedName("weather")
    public List<Weather> weathers;

    @SerializedName("base")
    public String base;

    @SerializedName("main")
    public Main main;

    @SerializedName("visibility")
    public String visibility;

    @SerializedName("wind")
    public Wind wind;

    @SerializedName("rain")
    public Rain rain;

    @SerializedName("snow")
    public Snow snow;

    @SerializedName("clouds")
    public Clouds clouds;

    @SerializedName("dt")
    public String dt;

    public class Coord {

        @SerializedName("lon")
        public String lon;
        @SerializedName("lat")
        public String lat;

    }

    public class Weather {

        @SerializedName("id")
        public String id;

        @SerializedName("main")
        public String main;

        @SerializedName("description")
        public String description;

        @SerializedName("icon")
        public String icon;

    }

    public class Main {

        @SerializedName("temp")
        public String temp;

        @SerializedName("pressure")
        public String pressure;

        @SerializedName("humidity")
        public String humidity;

        @SerializedName("temp_min")
        public String temp_min;

        @SerializedName("temp_max")
        public String temp_max;
    }

    public class Wind {

        @SerializedName("speed")
        public String speed;

        @SerializedName("deg")
        public String deg;

    }

    public class Rain {
        @Nullable
        @SerializedName("3h")
        public String rain;
    }

    public class Snow {
        @Nullable
        @SerializedName("3h")
        public String snow;
    }

    public class Clouds {
        @SerializedName("all")
        public String all;
    }

    public class Sys {
        @SerializedName("type")
        public String type;

        @SerializedName("id")
        public String id;

        @SerializedName("message")
        public String message;

        @SerializedName("country")
        public String country;

        @SerializedName("sunrise")
        public String sunrise;

        @SerializedName("sunset")
        public String sunset;

    }
}

File: kotlin/src/main/java/kr/co/jayhy/weather/repo/WeatherCurrentRepo.kt
package kr.co.jayhy.weather.repo

import com.google.gson.annotations.SerializedName

/**
 * Created by jhkim on 2017-08-06.
 */

class WeatherCurrentRepo {

    @SerializedName("cod")
    var cod: String? = null

    @SerializedName("coord")
    var coord: Coord? = null

    @SerializedName("weather")
    var weathers: List<Weather>? = null

    @SerializedName("base")
    var base: String? = null

    @SerializedName("main")
    var main: Main? = null

    @SerializedName("visibility")
    var visibility: String? = null

    @SerializedName("wind")
    var wind: Wind? = null

    @SerializedName("rain")
    var rain: Rain? = null

    @SerializedName("snow")
    var snow: Snow? = null

    @SerializedName("clouds")
    var clouds: Clouds? = null

    @SerializedName("dt")
    var dt: String? = null

    inner class Coord {

        @SerializedName("lon")
        var lon: String? = null
        @SerializedName("lat")
        var lat: String? = null

    }

    inner class Weather {

        @SerializedName("id")
        var id: String? = null

        @SerializedName("main")
        var main: String? = null

        @SerializedName("description")
        var description: String? = null

        @SerializedName("icon")
        var icon: String? = null

    }

    inner class Main {

        @SerializedName("temp")
        var temp: String? = null

        @SerializedName("pressure")
        var pressure: String? = null

        @SerializedName("humidity")
        var humidity: String? = null

        @SerializedName("temp_min")
        var temp_min: String? = null

        @SerializedName("temp_max")
        var temp_max: String? = null
    }

    inner class Wind {

        @SerializedName("speed")
        var speed: String? = null

        @SerializedName("deg")
        var deg: String? = null

    }

    inner class Rain {
        @SerializedName("3h")
        var rain: String? = null
    }

    inner class Snow {
        @SerializedName("3h")
        var snow: String? = null
    }

    inner class Clouds {
        @SerializedName("all")
        var all: String? = null
    }

    inner class Sys {
        @SerializedName("type")
        var type: String? = null

        @SerializedName("id")
        var id: String? = null

        @SerializedName("message")
        var message: String? = null

        @SerializedName("country")
        var country: String? = null

        @SerializedName("sunrise")
        var sunrise: String? = null

        @SerializedName("sunset")
        var sunset: String? = null

    }
}
